### DISCUSSION TITLE
Rocket import statement

### DISCUSSION THREAD

**Discussion by ekanna** [rwf2/Rocket#1700] [Questions] [answered]
Why this ?
```rust
#[macro_use] extern crate rocket;
```

Why not?
```rust
use rocket;
```

I am new to rust and trying to understand why you are using this pattern?

**Comment by paulzhang5511** [comment, marked as answer]
![11](https://user-images.githubusercontent.com/53463972/122008249-63c81e00-cdeb-11eb-8ff4-88b6249da0ea.png)

https://rocket.rs/v0.5-rc/guide/overview/#lifecycle

That's the official answer.
